app/assets/javascripts/features/machines.js

```diff
@@ -0,0 +1,49 @@
+const buttons = [...document.querySelectorAll('button')]
+const solutions = [...document.querySelectorAll('.problem__solutions')]
+const headers = [...document.querySelectorAll('.device__problems header')]
+
+var first_device = {
+  device: document.querySelector('.device__1'),
+  list: document.querySelector('.device-1__problems')
+}
+var second_device = {
+  device: document.querySelector('.device__2'),
+  list: document.querySelector('.device-2__problems')
+}
+var third_device = {
+  device: document.querySelector('.device__3'),
+  list: document.querySelector('.device-3__problems')
+}
+
+var devices = [first_device, second_device, third_device]
+
+const removeThis = i => {
+  devices.forEach(i => {
+    i.device.classList.remove('device-selected')
+    i.list.classList.add('__inactive')
+  })
+}
+
+const addThis = i => {
+  i.device.classList.add('device-selected')
+  i.list.classList.remove('__inactive')
+}
+
+devices.map(function(i) {
+  i.device.onclick = function() {
+    if (this.classList.contains('device-selected')) {
+    } else {
```

Comment:
You shouldn't have an empty `if`.  If you only want to do those when the condition is not true, that should look like:

```
if (!this.classList.contains('device-selected')) {
    removeThis(i)
    addThis(i)
}
```

☝️ note the `!` which means "not", so "if x is not true"